DuckDB.NET.Data/DuckDBDataReader.cs

```diff
@@ -191,17 +191,28 @@ public override string GetString(int ordinal)
 
     public override T GetFieldValue<T>(int ordinal)
     {
-        return vectorReaders[ordinal].ColumnDuckDBType switch
+        var value = vectorReaders[ordinal].ColumnDuckDBType switch
         {
-            DuckDBType.List => (T)vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, (dynamic)Activator.CreateInstance<T>()),
-            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1),
-            _ => (T)vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
+            DuckDBType.List => vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, Activator.CreateInstance<T?>() as dynamic),
+            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1) ,
+            _ => vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
         };
+
+        if (value is not null)
+            return (T)value;
+
+        if (default(T) is null && typeof(T).IsValueType)
+            return default!;
+
+        if (typeof(T) == typeof(object))
+            return (T)(object)DBNull.Value;
+
+        throw new InvalidCastException();
     }
 
     public override object GetValue(int ordinal)
     {
-        return IsDBNull(ordinal) ? DBNull.Value : vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1);
+        return IsDBNull(ordinal) ? DBNull.Value : vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1) ?? DBNull.Value;
```

Comment:
Will there be any case when `?? DBNull.Value` is executed?

Reply:
GetValue returns T? so in theory, yes it could be the case. I didn't check in detail if GetValue could effectively return null but I'd prefer to apply some encapsulation and not analyze the details of this method in case you've to change the details.

Reply:
The question is can we add null to a vector in DuckDB? If yes, then it makes sense. If it's not supported by DuckDB then vectorReaders[ordinal].GetValue(...) should not return a T? but a T.

Reply:
As far as I understand if a vector contains null, `IsDBNull(ordinal)` will return true